ui/pages/confirmations/hooks/alerts/useTokenTrustSignalAlerts.ts

```diff
@@ -0,0 +1,89 @@
+import { useMemo } from 'react';
+import {
+  SimulationTokenStandard,
+  TransactionMeta,
+} from '@metamask/transaction-controller';
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { RowAlertKey } from '../../../../components/app/confirm/info/row/constants';
+import { useConfirmContext } from '../../context/confirm';
+import { TrustSignalDisplayState } from '../../../../hooks/useTrustSignals';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+// eslint-disable-next-line import/no-restricted-paths
+import { isSecurityAlertsAPIEnabled } from '../../../../../app/scripts/lib/ppom/security-alerts-api';
+import { useTokenTrustSignalsForAddresses } from '../../../../hooks/useTokenTrustSignals';
+
+const EMPTY_ALERTS: Alert[] = [];
+const EMPTY_ACTIONS: Alert['actions'] = [];
+
+export function useTokenTrustSignalAlerts(): Alert[] {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext();
+
+  const txMeta = currentConfirmation as TransactionMeta | undefined;
+  const chainId = txMeta?.chainId;
+  const tokenBalanceChanges = txMeta?.simulationData?.tokenBalanceChanges;
+
+  const incomingTokenAddresses = useMemo<string[] | undefined>(() => {
+    if (!tokenBalanceChanges?.length) {
+      return undefined;
+    }
+    const dedup = new Set<string>();
+    for (const change of tokenBalanceChanges) {
+      if (
+        !change.isDecrease &&
+        change.standard === SimulationTokenStandard.erc20 &&
+        change.address
+      ) {
+        dedup.add(change.address.toLowerCase());
+      }
+    }
+    return dedup.size ? Array.from(dedup) : undefined;
+  }, [tokenBalanceChanges]);
```

Comment:
Do we need to dedup? Isn't that already handle by the simulation data? Can we have the same address multiple times?